ui/app/AppLayouts/Chat/views/ChatMessagesView.qml

```diff
@@ -151,7 +155,14 @@ Item {
             if(d.scrollY < 500) messageStore.loadMoreMessages()
         }
 
-        onCountChanged: d.markAllMessagesReadIfMostRecentMessageIsInViewport()
+        onCountChanged: {
+            if (!visible)
+                d.addNewMessagesMarker()
```

Comment:
I believe this is redundant to `onHasUnreadMessagesChanged()`, please note that the count change may eventually trigger `hasUnreadMessagesChanged`.

Could you please remove this part and adapt `onHasUnreadMessagesChanged` accordingly (remove unneeded conditions)?
```qml
function onHasUnreadMessagesChanged() {
    if (d.chatDetails.hasUnreadMessages) {
        // HACK: we call it later because messages model may not be yet propagated with unread messages when this signal is emitted
        Qt.callLater(() => messageStore.addNewMessagesMarker())
    }
}
```

Reply:
No, this is needed. 
This fix was done the previous week, so I do not remember all details for now but it was related to signal orders and you can mark messages as read before you'll add new messages marker.
If you'll remove it, you'll not see new message marker on a startup

Reply:
After your refactoring channel with a small amount of msgs:
https://user-images.githubusercontent.com/117639195/211367336-446b418b-b777-4799-b0e1-56434cf56693.mov

How it is now:
https://user-images.githubusercontent.com/117639195/211367471-eace7738-b523-4066-a8af-f758b268ee66.mov

Or check the attached video on the 24th second